_posts/2020-11-23-boost-your-productivity-with-tailwind-css.md

```diff
@@ -0,0 +1,184 @@
+
+* Purging of unused rules with PurgeCSS
+* Screenreader utilities
+* CSS transitions
+* CSS grid
+* Plugins
+* Animations
+* Dark mode
+
+## Examples
+
+Enough of the past. Let's get to the good stuff; Demos! We'll be using something called CodeSandbox, which is a fantastic tool, and I recommend it to you wholeheartedly. Unfortunately, you can not edit the embed below directly. If you wish to do that, push the "Open Sandbox" button and log in to CodeSandbox. The template I'm using uses Parcel for bundling, so it already includes PostCSS. To create the UI, the template features React v17.0.1 and Tailwind CSS v1.9.6. You could also use Angular, Svelte, Vue, or whatever for building components as well. If you want to quickly try out Tailwind CSS and don't want to hassle with JavaScript components, you could use [Tailwind Play](https://play.tailwindcss.com/).
+
+### Example 1 - Basics
+
+<iframe
+  src="https://codesandbox.io/embed/basics-boost-your-productivity-with-tailwind-css-xnbrm?autoresize=1&fontsize=14&hidenavigation=1&module=%2Fsrc%2FApp.js&moduleview=1&theme=dark&runonclick=1&codemirror=1&view=split&editorsize=66&highlights=5,6,7,8,9"
+    style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+    title="Basics | Boost your productivity with Tailwind CSS"
+    allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+    sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+></iframe>
+
+In example 1, we see Tailwind in its most basic form. It provides the button styles to make it look like a button. When the input is `class="p-2 leading-none text-white bg-purple-600 rounded-sm"` the output is
+
+```css
+.p-2 {
+  padding: 0.5rem;
+}
+.bg-purple-600 {
+  --bg-opacity: 1;
+  background-color: #805ad5;
+  background-color: rgba(128, 90, 213, var(--bg-opacity));
+}
+.leading-none {
+  line-height: 1;
+}
+.rounded-sm {
+  border-radius: 0.125rem;
+}
+.text-white {
+  --text-opacity: 1;
+  color: #fff;
+  color: rgba(255, 255, 255, var(--text-opacity));
+}
+```
+
+Typically you would give the element a single class, such as `class="btn"`, and then you would write in your CSS `.btn { ... }`. This example I've just introduced you is, of course, wildly exaggerated, but I want you to pay attention to the markup.
+
+```html
+<button class="p-2 leading-none text-white bg-purple-600 rounded-sm">Continue</button>
+<button class="btn">Continue</button>
+```
+
+Notice how it is immediately clearer how the button will look like just by looking at the markup. Utility-first CSS is all about these classes, how they are named, their output, and the connection between them. The classes are all reusable little snippets that provide functional styles, and since they already exist, you don't have to spend any effort to come up with fitting names.
```

Comment:
”Notice how it is immediately clearer how the button will look like just by looking at the markup.”

I think the main point of CSS is to separate presentation from markup. So mixing them is often considered kind of a bad thing.

Maybe you could add something to clarify why this approach would be a good thing? Why should we dump the whole ”separate presentation from the data” approach?

Reply:
I don't really want to get into all that, but maybe I could expand it little here. Maybe with a quote or something...